translate/src/main/java/com/google/cloud/translate/samples/TranslateText.java

```diff
@@ -0,0 +1,143 @@
+
+import com.google.cloud.RetryParams;
+import com.google.cloud.translate.Detection;
+import com.google.cloud.translate.Language;
+import com.google.cloud.translate.Translate;
+import com.google.cloud.translate.Translate.TranslateOption;
+import com.google.cloud.translate.TranslateOptions;
+import com.google.cloud.translate.Translation;
+import com.google.common.collect.ImmutableList;
+
+import java.io.PrintStream;
+import java.util.List;
+
+public class TranslateText {
+  /**
+   * Detect the language of input text.
+   *
+   * @param sourceText source text to be detected for language
+   * @param out print stream
+   */
+  public static void detectLanguage(String sourceText, PrintStream out) {
+    Translate translate = createTranslateService();
+    List<Detection> detections = translate.detect(ImmutableList.of(sourceText));
+    System.out.println("Language(s) detected:");
+    for (Detection detection : detections) {
+      out.printf("\t%s\n", detection);
+    }
+  }
+
+  /**
+   * Translates the source text in any language to english.
+   *
+   * @param sourceText source text to be translated
+   * @param out print stream
+   */
+  public static void translateText(String sourceText, PrintStream out) {
+    Translate translate = createTranslateService();
+    Translation translation = translate.translate(sourceText);
+    out.printf("Source Text:\n\t%s\n", sourceText);
+    out.printf("Translated Text:\n\t%s\n", translation.translatedText());
+  }
+
+  /**
+   * Translate the source text from source to target language.
+   *
+   * @param sourceText source text to be translated
+   * @param sourceLang source language of the text
+   * @param targetLang target language of translated text
+   * @param out print stream
+   */
+  public static void translateTextWithOptions(
+      String sourceText,
+      String sourceLang,
+      String targetLang,
+      PrintStream out) {
+
+    Translate translate = createTranslateService();
+    TranslateOption srcLang = TranslateOption.sourceLanguage(sourceLang);
+    TranslateOption tgtLang = TranslateOption.targetLanguage(targetLang);
+
+    Translation translation = translate.translate(sourceText, srcLang, tgtLang);
+    out.printf("Source Text:\n\tLang: %s, Text: %s\n", sourceLang, sourceText);
+    out.printf("TranslatedText:\n\tLang: %s, Text: %s\n", targetLang, translation.translatedText());
+  }
+
+  /**
+   * Displays a list of supported languages and codes.
+   *
+   * @param out print stream
+   */
+  public static void displaySupportedLanguages(PrintStream out) {
+    Translate translate = createTranslateService();
+    List<Language> languages = translate.listSupportedLanguages();
+
+    for (Language language : languages) {
+      out.printf("Name: %s, Code: %s\n", language.name(), language.code());
+    }
+  }
+
+  /**
+   * Create Google Translate API Service.
+   *
+   * @return Google Translate Service
+   */
+  public static Translate createTranslateService() {
```

Comment:
Are the default options not sufficient that this function is needed? Please file a bug against google-cloud-java if the defaults aren't working well for you.